hvpy/utils.py

```diff
@@ -164,9 +166,18 @@ def save_file(data: bytearray, filename: Union[Path, str], overwrite: bool = Fal
     overwrite
         Whether to overwrite the file if it already exists.
         Default is `False`.
+
+    Returns
+    -------
+    `~pathlib.Path`
+        The path to the saved file.
     """
-    if isinstance(filename, str):
-        filename = Path(filename)
+    filename = Path(filename).expanduser().resolve().absolute()
+    filepath, filename = os.path.split(filename)
+    # Sanitize the filename - Only works for strings
+    filename = re.sub(r"[^\w\-_\. ]", "_", filename)
+    filename = Path(filepath) / Path(filename)
     if filename.exists() and not overwrite:
         raise FileExistsError(f"{filename} already exists. Use overwrite=True to overwrite.")
     filename.write_bytes(data)
+    return filename
```

Comment:
Was there a reason to not do this?

Reply:
Actually, I don't recall. It is better now.